R/Connection.R

```diff
@@ -712,6 +712,53 @@ setMethod(
     invisible(TRUE)
   })
 
+#' List locations of ODBC configuration files
+#'
+#' @description
+#' On MacOS and Linux, odbc uses the unixODBC driver manager to manage
+#' information about driver and data sources. This helper returns the filepaths
+#' where the driver manager will look for that information.
+#'
+#' This function is a wrapper around the command line call `odbcinst -j`.
+#'
+#' Windows does not use `.ini` configuration files; this function will return a
+#' 0-length vector on Windows.
+#'
+#' @seealso
+#' The [odbcListDrivers()] and [odbcListDataSources()] helpers return
+#' information on the contents of `odbcinst.ini` and `odbc.ini` files,
+#' respectively.
+#'
+#' Learn more about unixODBC and the `odbcinst` utility
+#' [here](https://www.unixodbc.org/odbcinst.html).
+#'
+#' @examplesIf FALSE
+#' configs <- odbcListConfig()
+#'
+#' file.edit(configs[1])
+#' @export
+odbcListConfig <- function() {
+  if (identical(.Platform$OS.type, "windows")) {
+    return(character(0))
+  }
+
+  if (identical(unname(Sys.which("odbcinst")), "")) {
```

Comment:
I think it'd be worth extracting this out to a new `has_odbc()` function. That'll make the skips easier and you can also add a snapshot test of this error by mocking `has_odbc()` to return `FALSE`.

Reply:
[275b69c](https://github.com/r-dbi/odbc/pull/680/commits/275b69c3740b695d8b96fb6e2d265d81553c1fa6)!